### PR TITLE
Gnuport cleanup 3

### PR DESCRIPTION
**Pull request by franksinankaya**
- Abstract __declspec(noinline)
- Abstract __declspec(nothrow)
- Need double quotes around single quotes in error statements
- Reorder extern "C" and visibility statements

### CODE REVIEW

File: src/vm/crst.h [hunk 1]
```diff
@@ -106,6 +106,14 @@ extern DWORD g_fEEShutDown;
 extern Volatile<LONG> g_ShutdownCrstUsageCount;
 extern Volatile<LONG> g_fForbidEnterEE;
 
+#ifndef NOINLINE
```
Comment (jkotas) [inline src/vm/crst.h]: We should define it once e.g. somewhere around here: https://github.com/dotnet/coreclr/blob/master/src/inc/palclr.h#L468

Comment (franksinankaya) [reply src/vm/crst.h]: done. Agree it is much more cleaner now.

File: src/inc/cor.h [hunk 2]
```diff
@@ -2026,16 +2026,24 @@ typedef enum
 
 #ifndef DEBUG_NOINLINE
 #if defined(_DEBUG)
+#ifdef _MSC_VER
 #define DEBUG_NOINLINE __declspec(noinline)
 #else
+#define DEBUG_NOINLINE __attribute__((noinline))
```
Comment (jkotas) [inline src/inc/cor.h]:
```suggestion
#define DEBUG_NOINLINE NOINLINE
```

File: src/inc/cor.h [hunk 3]
```diff
@@ -2026,16 +2026,24 @@ typedef enum
 
 #ifndef DEBUG_NOINLINE
 #if defined(_DEBUG)
+#ifdef _MSC_VER
```
Comment (jkotas) [inline src/inc/cor.h]: I meant replace this ifdef with just `#define DEBUG_NOINLINE NOINLINE`

File: src/inc/cor.h [hunk 4]
```diff
@@ -2026,16 +2026,24 @@ typedef enum
...
+#define DEBUG_NOINLINE NOINLINE
+#endif
+#else
 #define DEBUG_NOINLINE
 #endif
 #endif
 
 #ifndef DBG_NOINLINE_X86__RET_INLINE
 #if defined(_DEBUG) && defined(_X86_)
+#ifdef _MSC_VER
```
Comment (jkotas) [inline src/inc/cor.h]: Similar here

File: src/inc/utilcode.h [hunk 5]
```diff
@@ -144,17 +144,25 @@ typedef LPSTR   LPUTF8;
...
+#else
 #define DEBUG_NOINLINE
 #endif
 #endif
 
 #ifndef DBG_NOINLINE_X86__RET_INLINE
 #if defined(_DEBUG) && defined(_TARGET_X86_)
 // this exists to make scan work on x86. 
+#ifdef _MSC_VER
 #define DBG_NOINLINE_X86__RET_INLINE __declspec(noinline)
```
Comment (jkotas) [inline src/inc/utilcode.h]: Actually, I think you can just delete this weird macro. It does not seem to be used anywhere.

**Comment by franksinankaya** [comment]
@dotnet-bot test Windows_NT x64 Checked CoreFX Tests 
@dotnet-bot test  coreclr-ci  
@dotnet-bot test  coreclr-ci (Test pri0 Linux_musl x64 checked)

**Comment by franksinankaya** [comment]
not fun

@dotnet-bot test Windows_NT x64 Checked CoreFX Tests
@dotnet-bot test coreclr-ci
@dotnet-bot test coreclr-ci (Test pri0 Linux_musl x64 checked)

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**